VisualizationBase/src/items/ViewItem.cpp

```diff
@@ -274,4 +278,87 @@ void ViewItem::ensureColumnExists(int column)
 		nodes_.resize(column + 1);
 }
 
+//JSON storing/loading
+QJsonDocument ViewItem::toJson() const
+{
+	//Store all the nodes
+	QJsonArray nodes;
+	for (int col = 0; col < nodes_.size(); col++)
+		for (int row = 0; row < nodes_[col].size(); row++)
+			if (auto node = DCast<ViewItemNode>(nodes_[col][row]))
+				nodes.append(node->toJson(this));
+	//Store all the arrows
+	QJsonArray arrows;
+	for (auto key : arrows_.keys())
+		for (auto pair : arrows_[key])
+			arrows.append(arrowToJson(pair, key));
+	QJsonObject main;
+	main.insert("nodes", nodes);
+	main.insert("arrows", arrows);
+	main.insert("name", name());
+	QJsonDocument result;
+	result.setObject(main);
+	return result;
+}
+
+void ViewItem::fromJson(QJsonDocument json)
+{
+	nodes_.clear();
+	arrows_.clear();
+	QJsonObject obj = json.object();
+	setName(obj["name"].toString());
+	QStringList steps{"NODE", "SPACING", "INFO"};
+	int step = 0;
+	auto objArray = obj["nodes"].toArray();
+	//Load the nodes: First normal nodes, then spacing nodes, and lastly info nodes
+	for (step = 0; step < steps.size(); step++)
+		for (int i = 0; i < objArray.size(); i++)
+		{
+			auto current = objArray[i].toObject();
+			if (current["type"] == steps[step])
+				insertViewItemNode(ViewItemNode::withJson(current, this),
+								   current["col"].toInt(), current["row"].toInt());
+		}
+	//Load the arrows
+	auto arrowArray = obj["arrows"].toArray();
+	for (auto arrow : arrowArray)
+		arrowFromJson(arrow.toObject());
+}
+
+QJsonObject ViewItem::arrowToJson(QPair<Item*, Item*> arrow, QString layer) const
+{
+	QJsonObject json;
+	json.insert("layer", layer);
+	//The first/second nodes might be a ViewItemNode -> in that case handled by the parent
+	if (arrow.first->node()->manager())
+		json.insert("node1", arrow.first->node()->manager()->
+								 nodeIdMap().id(arrow.first->node()).toString());
+	if (arrow.second->node()->manager())
+		json.insert("node2", arrow.second->node()->manager()->
+								 nodeIdMap().id(arrow.second->node()).toString());
+	auto parent1 = arrow.first->findAncestorOfType<VViewItemNode>();
+	json.insert("parent1col", positionOfItem(parent1).x());
```

Comment:
Kind of. For example, right now a method might be displayed multiple times, but the calling method does not have arrows to all of them (as not all of them were added in the context of showing it as a callee of the other method).

I feel this approach is good, since we need almost no case distinction, and it will always restore the arrows exactly as they were.